File: 编译原理/TinyScript-Js/src/parser/ast/DeclareStmt.js
const Stmt = require('./Stmt')
const ASTNodeTypes = require('./ASTNodeTypes')

/**
 * 抽象语法树 - 声明语句节点
 */
class DeclareStmt extends Stmt {
  /**
   * @param {ASTNode} _parent
   * @param {ASTNodeTypes} _type
   * @param {string} _label
   */
  constructor(_parent) {
    super(_parent, ASTNodeTypes.DECLARE_STMT, 'declare')
  }
}

module.exports = DeclareStmt

File: 编译原理/TinyScript/src/main/java/parser/ast/DeclareStmt.java
package parser.ast;

/**
 * 抽象语法树 - 声明语句节点
 */
public class DeclareStmt extends Stmt {
    public DeclareStmt(ASTNode _parent) {
        super(_parent, ASTNodeTypes.DECLARE_STMT, "declare");
    }
}
